### PR TITLE
gopfmt: support classfile

### PR DESCRIPTION
**Pull request by visualfc**
gop fmt and gopfmt support lookup classfile
- support load classfile from gopmod
- support classfile embed type
- add .gopx

### CODE REVIEW

File: cmd/internal/gopfmt/fmt.go [hunk 1]
```diff
@@ -127,6 +123,10 @@ func writeFileWithBackup(path string, target []byte) (err error) {
 	return os.Rename(tmpfile, path)
 }
 
+var (
+	dirMap = make(map[string]func(ext string) (ok bool, class bool))
```
Comment (xushiwei) [inline cmd/internal/gopfmt/fmt.go:127]: don't use global variables. use a `walker` object:

```go
type walker struct {
    dirMap map[string]func(ext string) (ok, class bool)
}

func (w walker) walk(path string, d fs.DirEntry, err error) error {
    ...
}
```

File: load.go [hunk 2]
```diff
@@ -51,7 +51,7 @@ func NotFound(err error) bool {
 	return errors.Err(err) == syscall.ENOENT
 }
 
-func loadMod(dir string, gop *env.Gop, conf *Config) (mod *gopmod.Module, err error) {
+func LoadMod(dir string, gop *env.Gop, conf *Config) (mod *gopmod.Module, err error) {
 	mod, err = gopmod.Load(dir, 0)
 	if err != nil && !NotFound(err) {
 		err = errors.NewWith(err, `gopmod.Load(dir, 0`, -2, "gopmod.Load", dir, 0)
```
Comment (xushiwei) [inline load.go:57]:
```go
errors.NewWith(err, `gopmod.Load(dir, 0`
```
misses `)`:

```go
errors.NewWith(err, `gopmod.Load(dir, 0)`
```

File: cmd/internal/gopfmt/fmt.go [hunk 3]
```diff
@@ -135,15 +135,46 @@ func walk(path string, d fs.DirEntry, err error) error {
...
 	} else {
+		dir, _ := filepath.Split(path)
+		fn, ok := dirMap[dir]
+		if !ok {
+			if mod, err := gop.LoadMod(path, nil, &gop.Config{DontUpdateGoMod: true}); err == nil {
+				fn = func(ext string) (ok bool, class bool) {
+					switch ext {
+					case ".go", ".gop":
+						ok = true
+					case ".gopx", ".spx", "gmx":
```
Comment (xushiwei) [inline cmd/internal/gopfmt/fmt.go:146]: ".gmx"

File: cmd/internal/gopfmt/fmt.go [hunk 4]
```diff
@@ -135,15 +135,46 @@ func walk(path string, d fs.DirEntry, err error) error {
...
+						}
+					}
+					return
+				}
+			} else {
+				fn = func(ext string) (ok bool, class bool) {
+					switch ext {
+					case ".go", ".gop":
+						ok = true
+					case ".gopx", ".spx", "gmx":
```
Comment (xushiwei) [inline cmd/internal/gopfmt/fmt.go:161]: ".gmx"

File: cmd/internal/gopfmt/fmt.go [hunk 5]
```diff
@@ -135,15 +135,46 @@ func walk(path string, d fs.DirEntry, err error) error {
...
+						ok, class = true, true
+					default:
+						_, class = mod.IsClass(ext)
+						if class {
+							ok = true
+						}
+					}
+					return
+				}
+			} else {
```
Comment (xushiwei) [inline cmd/internal/gopfmt/fmt.go:156]: don't need use `if..else`

File: cmd/internal/gopfmt/fmt.go (hunk 1 above)
Comment (visualfc) [reply cmd/internal/gopfmt/fmt.go]: fixed

**Pull request merged**